modules/claims_api/app/controllers/concerns/claims_api/disability_compensation_validations.rb

```diff
@@ -232,6 +234,19 @@ def validate_form_526_veteran_homelessness!
       end
     end
 
+    def validate_service_after_13th_birthday!
+      age_thirteen = auth_headers&.dig('va_eauth_birthdate')&.to_datetime&.next_year(13)&.to_time
+      started_after_age_thirteen = form_attributes&.dig('serviceInformation', 'servicePeriods')&.any? do |period|
+        period&.dig('activeDutyBeginDate')&.< age_thirteen
```

Comment:
Instead of comparing string dates lexicographically, would it be safer to compare two `Date` objects? i.e.,

```rb
age_thirteen = auth_headers.dig('va_eauth_birthdate').to_date
#...
Date.parse(period.dig('activeDutyBeginDate')) < age_thirteen
```
This way, if any dates are improperly formatted, we raise an error, rather than potentially compare invalid strings?